Word 365: ignore UIA notification events seen after performing various editing functions

josephsl opened the pull request:
### Link to issue number:
 Fixes #10851 

### Summary of the issue:
In Word 365, word document UIA object fires UIA notification events after editing functions are performed. For example, NVDA will say "delete back word" when Control+Backspace is pressed.

### Description of how this pull request fixes the issue:
Added a "dummy" UIA notification event handler in NVDAObjects.UIA.wordDocument.WordDocument class that simply returns, thus making NVDA ignore UIA notification events.

### Testing performed:
Tested with Word 365, NVDA source code, and a somewhat limited fix in Windows 10 App Essentials add-on.

### Known issues with pull request:
None

### Change log entry:
Bug fixes:

In recent releases of Microsoft Word 365, NVDA will no longer announce "delete back word" when Control+Backspace is pressed while editing a document.

josephsl committed:
NVDAObjects/UIA/WordDocument: update copyright header

josephsl committed:
Word document UIA object: ignore UIA notification event. Re #10851.

In recent Word 365 releases, UIA notification events are fired by Word document object when editing functions are performed such as 'delete back word' being announced when Control+Backspace is pressed. Thus ignore these messages.

LeonarddeR reviewed (dismissed):
I"m afraid this will also kill useful notifications, such as when someone starts collaborating with you on a document

josephsl commented:
Hi, true, hence the quick fix for now. I’m sure we can filter them out based on activity ID. Thanks.

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Monday, March 9, 2020 6:45 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Word 365: ignore UIA notification events seen after performing various editing functions (#10857)

 

@leonardder requested changes on this pull request.

I"m afraid this will also kill useful notifications, such as when someone starts collaborating with you on a document

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10857?email_source=notifications&email_token=AB4AXEESO5TGAEQTKWSBVH3RGTXFXA5CNFSM4LEBRQCKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCYP7Q3A#pullrequestreview-371193964> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEBHUXEFOG5TGWCYHXDRGTXFXANCNFSM4LEBRQCA> .

josephsl committed:
Word document UIA notification: only suppress delete function announcements. Re #10851.

Reviewed by Leonard de Ruijter: there are messages in Word 365 that are announced using UIA notification. Therefore only suppress delete messages (activity ID: AccSN2).

feerrenrut reviewed (commented):
I have tested this and confirmed that "delete back word" is no longer reported. I did not test that other UIA notifications are still reported.

@josephsl Could you comment on any testing you have done to that effect?

josephsl commented:
Hi, I used Windows 10 App Essentials add-on/debug output mode to confirm that: the activity ID we are looking for is indeed the ones that should be silenced, and other operations such as cut/copy/paste are announced as expected. This also works without the add-on present (source code copy). Thanks.

 

From: Reef Turner <notifications@github.com> 
Sent: Friday, March 27, 2020 9:38 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [nvaccess/nvda] Word 365: ignore UIA notification events seen after performing various editing functions (#10857)

 

@feerrenrut commented on this pull request.

I have tested this and confirmed that "delete back word" is no longer reported. I did not test that other UIA notifications are still reported.

@josephsl <https://github.com/josephsl>  Could you comment on any testing you have done to that effect?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10857#pullrequestreview-383022936> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEBI3QFSGLR6RQ23OW3RJTI7JANCNFSM4LEBRQCA> .

feerrenrut reviewed (approved):
Thanks @josephsl
